docs/assets.md

```diff
@@ -0,0 +1,151 @@
+# Asset Management in Whitehall
+
+*Assets, Attachments, Images, Logos, Featured Image, Lead Image, Response Form*
+
+# What is it
+
+There are several features in Whitehall that utilize Images or
+File Attachments.
+
+For example News Articles can have a list of attachments that
+provide further information about the news. Or a Person can have a picture
+which is shown on the person's page and also on any Speech page by that person.
+
+# How did it use to work
+
+Historically Whitehall had all file uploads stored within the Whitehall service. 
+Carrierwave was used to manage the http requests and related file storage within
+Whitehall.
+
+Later a separate service - Asset Manager - became the store for file uploads, taking ownership of every file uploaded.
+Carrierwave was still used to mange the http-request in Whitehall and a plentiful amount of configuration was
+written for carrierwave in order to make AssetManager behave as if it was just a file storage.
+These configurations rippled throughout Whitehall codebase and specialized its behaviour in
+Asset Manager, locking asset manager design and making the overhead of understanding Whitehall
+very heavy. Especially "Legacy url path" was a concept that heavily blocked any refactoring attempts.
+
+Later, the "Legacy url path" was removed and Whitehall Asset references became just simple rows in the
+database.
+
+Please find a related ADR for the asset model [here](adr/0002-new-asset-model.md)
+
+
+# Technical Design
+Most content types utilise attachments or images one way or other. 
+
+![Document types that use Attachments and Images](diagrams/asset_business_domains.png)
+
+![Model diagram](diagrams/asset_models.png)
+
+### Attachments upload flow
```

Comment:
do we also want to add a separate description for forms?

Reply:
i feel like it's unnecessary to include all the things - forms are effectively just file attachments, just like all the other things